### PR TITLE
ddl,meta: Generate global IDs in bulk to reduce the number of lock acquisition

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

When generating global IDs for partitioned tables, the underlying `(*Meta).GenGlobalID`
was called repeatedly, acquiring and releasing the `globalIDMutex`.

### What is changed and how it works?

Add a method `(*Meta).GenGlobalIDs` that generate multiple global IDs at
once.

### Check List 

Tests 

 - Unit test

Code changes

 - Has interface methods change

Side effects

Related changes

### CODE REVIEW

File: meta/meta.go [hunk 1]
```diff
@@ -117,6 +117,23 @@ func (m *Meta) GenGlobalID() (int64, error) {
 	return m.txn.Inc(mNextGlobalIDKey, 1)
 }
 
+// GenGlobalIDs generates the next n global IDs.
+func (m *Meta) GenGlobalIDs(n int) ([]int64, error) {
+	globalIDMutex.Lock()
+	defer globalIDMutex.Unlock()
+
+	new, err := m.txn.Inc(mNextGlobalIDKey, int64(n))
```
Comment (tiancaiamao) [inline meta/meta.go]: new is not a good variable name

File: meta/meta.go [hunk 2]
```diff
@@ -117,6 +117,23 @@ func (m *Meta) GenGlobalID() (int64, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	orig := new - int64(n)
```
Comment (tiancaiamao) [inline meta/meta.go]: If there are several tidb instance genGlobalIDs simultaneously, could this condition hold?

File: meta/meta.go (hunk 1 above)
Comment (suzaku) [reply meta/meta.go]: Updated to `newID`.

File: meta/meta.go (hunk 2 above)
Comment (suzaku) [reply meta/meta.go]: Since we are holding the lock, this should be OK. I also modify the test case to make the two `GenGlobalIDs` call concurrently.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**